src/datastore/pg-store-v2.ts

```diff
@@ -230,4 +234,40 @@ export class PgStoreV2 extends BasePgStoreModule {
       if (blockQuery.count > 0) return blockQuery[0];
     });
   }
+
+  async getSmartContractStatus(args: SmartContractStatusParams): Promise<DbSmartContractStatus[]> {
+    return await this.sqlTransaction(async sql => {
+      const statusArray: DbSmartContractStatus[] = [];
+      const contractArray = Array.isArray(args.contract_id) ? args.contract_id : [args.contract_id];
+
+      // Search confirmed txs.
+      const confirmed = await sql<DbSmartContractStatus[]>`
+        SELECT DISTINCT ON (smart_contract_contract_id) smart_contract_contract_id, tx_id, block_height, status
+        FROM txs
+        WHERE type_id IN ${sql([DbTxTypeId.SmartContract, DbTxTypeId.VersionedSmartContract])}
+          AND smart_contract_contract_id IN ${sql(contractArray)}
+          AND canonical = TRUE
+          AND microblock_canonical = TRUE
+          AND status = ${DbTxStatus.Success}
```

Comment:
Not sure if I should be doing this condition @zone117x because it would be useful to know if a contract failed or something (even if it's not in the mempool anymore). Thoughts?

Reply:
Maybe you could instead add the `status` to the ORDER BY clause, that way for a given contract, you'll get either the successful deployment, or the last failed one?

Reply:
Great idea, I'll do that